src/goal_gui.cpp

```diff
@@ -488,5 +578,5 @@ static WindowDesc _goal_question_list_desc(
  */
 void ShowGoalQuestion(uint16 id, byte type, uint32 button_mask, const char *question)
 {
-	new GoalQuestionWindow(&_goal_question_list_desc, id, type, button_mask, question);
+	new GoalQuestionWindow(&_goal_question_list_desc[type], id, type == 3 ? TC_WHITE : TC_BLACK, button_mask, question);
```

Comment:
given `type` comes in as a byte, would you mind adding an `assert` to ensure it is within the range of the array?

Reply:
In fact, it used to be there:

```
assert(type < GOAL_QUESTION_TYPE_COUNT);
```

but you seem to only have removed it :P :D